src/node/net.ts

```diff
@@ -1110,10 +1110,21 @@ async function startRead(socket: SocketClass): Promise<void> {
   const reader = socket._handle.reader;
   try {
     while (socket._handle.reading === true) {
+      const generatedBuffer = socket[kBufferGen]?.();
+
+      if (
+        generatedBuffer instanceof Uint8Array &&
+        generatedBuffer.length === 0
```

Comment:
I don't know this code well enough but would something like that work?

```suggestion
        generatedBuffer?.length === 0
```

That is, I can not tell if `generatedBuffer instanceof Uint8Array` is ok or too restrictive here?

Reply:
hmm... actually, if `generatedBuffer` can be `null` or `boolean` then this is still buggy as we can't pass `null` or `boolean` as the argument to `read()`

Reply:
Type-wise it can be null or boolean, but it will never be called with null or boolean. If it did Node.js implementation would not work. I'll make sure to handle null or boolean in here as well.